rust/lsp-lib/src/language_server_client.rs

```diff
@@ -98,38 +84,36 @@ impl LanguageServerClient {
     }
 
     pub fn handle_message(&mut self, message: &str) {
-        let value = JsonRpc::parse(message).unwrap();
 
-        match value {
-            JsonRpc::Request(obj) => eprintln!("client received unexpected request: {:?}", obj),
-            JsonRpc::Notification(obj) => eprintln!("\n\n recv notification: {:?} \n\n", obj),
-            JsonRpc::Success(_) => {
-                let mut result = value.get_result().unwrap().to_owned();
-                let id = number_from_id(value.get_id().as_ref());
-                self.handle_response(id, result);
-            }
-            JsonRpc::Error(_) => {
-                let mut error = value.get_error().unwrap().to_owned();
-                let id = number_from_id(value.get_id().as_ref());
-                self.handle_error(id, error);
+        match JsonRpc::parse(message) {
+            Ok(value) => {
```

Comment:
Another observation: instead of having two match blocks, you can match on the specific results, e.g.:

```rust
match JsonRpc::parse(message) {
    Ok(JsonRpc::Request(obj)) => (),
    Ok(JsonRpc::Success(_)) => (),
    Err(err) => (),
}
```

Anyway this isn't blocking, but is a nice Rust thing.

Reply:
I need `value` in the inner match block. Is there a way to access it if I use the above format?

Reply:
Found the way. Using `@` notation